CMakeLists.txt

```diff
@@ -1,35 +1,46 @@
-cmake_minimum_required(VERSION 3.16.3)
+cmake_minimum_required(VERSION 3.16.3...3.25)
 # -- v3.16.3 is the default version in the current ubuntu lts release
 
-project(azmq LANGUAGES CXX)
+project(azmq VERSION 1.1.0 LANGUAGES CXX)
+
+if (PROJECT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+    set(CMAKE_MASTER_PROJECT YES)
+endif ()
 
 # -- options
-option(BUILD_TESTING "run unit tests" no)
+option(AZMQ_BUILD_TESTS "run unit tests" ${CMAKE_MASTER_PROJECT})
 option(BUILD_SHARED_LIBS "use the libzmq static or shared" yes)
-option(MORE_WARNINGS "More warnings" no)
+option(MORE_WARNINGS "More warnings" ${CMAKE_MASTER_PROJECT})
 option(WARNINGS_AS_ERRORS "Treat compiler warnings as errors" no)
 
 # -- warning and errors template from from
 # https://github.com/cpp-best-practices/cpp_starter_project/tree/main/cmake
-add_library(project_warnings INTERFACE)
-if (MORE_WARNINGS)
-    include(cmake/CompilerWarnings.cmake)
-    set_project_warnings(project_warnings)
+if (CMAKE_SKIP_INSTALL_RULES)
+    add_library(project_warnings INTERFACE)
+    if (MORE_WARNINGS)
+        include(cmake/CompilerWarnings.cmake)
+        set_project_warnings(project_warnings)
+    endif ()
+
+    add_library(project_options INTERFACE)
+    include(cmake/Sanitizers.cmake)
+    enable_sanitizers(project_options)
 endif ()
```

Comment:
This is for developers only, it must not installed!

Reply:
I agree it probably doesn't make sense to install the extra "build flag" targets that enable 'more warnings'.  Maybe these should get pushed down into the test project, so we only compile with these options when building the unit test.  Uses of the project can the decide on there own flags.